### PR TITLE
Migrate worker and starter to use Micrometer + Prometheus

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

Migrate worker and starter to use Micrometer + Prometheus.

## Related issues

relates to https://github.com/camunda/zeebe/issues/13914

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by rodrigo-lourenco-lopes** [comment]
Benchmark created: https://github.com/camunda/zeebe/actions/runs/5975695180
Grafana: https://grafana.dev.zeebe.io/d/zeebe-dashboard/zeebe?orgId=1&var-DS_PROMETHEUS=Prometheus&var-cluster=All&var-namespace=rl-test-micrometer2&var-pod=All&var-partition=All

After port fowarding the worker locally, we can see the metrics invrementing with the number of jobs:
Screenshot 2023-08-25 at 14 25 31

File: benchmarks/project/src/main/java/io/camunda/zeebe/Worker.java [hunk 1]
```diff
@@ -44,8 +46,12 @@ public void run() {
     final var variables = readVariables(workerCfg.getPayloadPath());
     final BlockingQueue<Future<?>> requestFutures = new ArrayBlockingQueue<>(10_000);
     final BlockingDeque<DelayedCommand> delayedCommands = new LinkedBlockingDeque<>(10_000);
-
     final ZeebeClient client = createZeebeClient();
+    final JobWorkerMetrics metrics =
+        JobWorkerMetrics.micrometer()
+            .withMeterRegistry(prometheusRegistry)
+            .withTags(Tags.of("WorkerName", workerCfg.getWorkerName(), "JobType", jobType))
```
Comment (npepinpe) [inline benchmarks/project/src/main/java/io/camunda/zeebe/Worker.java:53]: :x: Let's keep the convention for labels to be all lower case, so `workerName`, `jobType`, etc.

**Review by npepinpe — APPROVED** [review]
:rocket: 

Just one small thing, but no need for another review

Can you add a follow issue to update the dashboard to add client and job worker metrics?

**Comment by rodrigo-lourenco-lopes** [comment]
bors merge

**Pull request merged**